### ISSUE TITLE
Repository history regularly being rewritten

### ISSUE THREAD

**Issue by tarsius** [alexluigit/dirvish#123]
I maintain the [Emacsmirror](https://github.com/emacsmirror), which mirrors most emacs packages available and extracts metadata from all these packages.

When the history of a package's history is rewritten, then updating is suspended until I manually inspect what has happened. Usually what happened is that the author was not happy with the last or a very recent commit and decided to amend it. That is okay; I do it myself occasionally. I would however recommend to only do it if you notice the mistake within, say, an hour or two.

You are doing that occasionally, and as I usually do, I then quietly instruct the Emacsmirror tools to go with the new history.

But a few times now you have either rewritten all history or at least a few hundred commits. Usually I contact the author immediately when that happens in case they did it by mistake and would like to revert once they learn about it, but I am currently on a break, so I didn't.

Do you do this on purpose, and if so why? Do you update all tags to point at commits in the new history?

**Comment by alexluigit** [comment]
Sorry about that! I didn't realize it would cause these extra works to you, my bad.

> I would however recommend to only do it if you notice the mistake within, say, an hour or two.

Definitely.  I did that just because fixup + rebase saves a commit message writing, I'll adjust my workflow accordingly.

> Do you do this on purpose, and if so why?

Yes, if my memory serves me, this happened 2 or 3 times in this repo.  The reason is that I brought in a few images as screenshots in the earlier versions, which makes the file size of this repo unnecessarily huge (~40m).  All my history rewritten operations are attempts to remove them (at least in main branch so that a regular `git clone` won't download those blobs), and apparently I failed to get it right until the most recent one.  I don't think I have any other reasons to do that again.

> Do you update all tags to point at commits in the new history?

Not yet, but will do now :)

Thanks for all your work! Sorry again for the troubles I made!
